projects/storefrontlib/shared/components/generic-link/generic-link.models.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Pattern matching string starting with `http://` or `https://`.
+ */
+export const HTTP_PROTOCOL_REGEX: RegExp = /^https?:\/\//i;
+
+/**
+ * Pattern matching string starting with `mailto:`.
+ */
+export const MAILTO_PROTOCOL_REGEX: RegExp = /^mailto:/i;
+
+/**
+ * Pattern matching string starting with `tel:`.
+ */
+export const TEL_PROTOCOL_REGEX: RegExp = /^tel:/i;
```

Comment:
Nitpick: now those items are loose members of public API of `@spartacus/storefront`  and customers may use  them freely in various places in their customisations, e.g `import { TEL_PROTOCOL_REGEX } from '@spartacus/storefront'`. That's not too bad. But at the same time, I'm not convinced if it's necessary to expose them as loose items.

What do you think about moving those consts to protected fields of the service, so they are scoped only to the service's logic?